nni/compression/pytorch/speedup/quantization_speedup/integrated_tensorrt.py

```diff
@@ -0,0 +1,172 @@
+        self.input_shape = input_shape
+        self.config = config
+        self.extra_layer_bit = extra_layer_bit
+        self.strict_datatype = strict_datatype
+        self.using_calibrate = using_calibrate
+        self.calibrate_type = calibrate_type
+        self.calib_data = calib_data
+        self.calibration_cache = calibration_cache
+        self.batchsize = batchsize
+        self.input_names = input_names
+        self.output_names = output_names
+        self.context = None
+        self.onnx_config = {}
+
+    def build(self):
+        """
+        Get onnx config and build tensorrt engine.
+        """
+        assert self.model is not None
+        assert self.onnx_path is not None
+        assert self.input_shape is not None
+
+        # Convert pytorch model to onnx model and save onnx model in onnx_path
+        _, self.onnx_config = fonnx.torch_to_onnx(self.model, self.config, input_shape=self.input_shape, model_path=self.onnx_path, input_names=self.input_names, output_names=self.output_names)
+
+        if self.using_calibrate:
+            assert self.calibrate_type is not None
+            context = self.tensorrt_build_withcalib(self.onnx_path)
+        else:
+            raise NameError('quantized without calibrate has not been supported.')
+        self.context = context
+
+    def tensorrt_build_withcalib(self, onnx_path):
+        calib = calibrator.Calibrator(self.calib_data, self.calibration_cache, self.batchsize, self.calibrate_type)
+        engine = build_engine(onnx_path, calib, self.batchsize, self.onnx_config, self.extra_layer_bit, self.strict_datatype)
+        return engine.create_execution_context()
+
+    def inference(self, test_data):
+        """
+        Do inference by tensorrt builded engine.
+        """
+        # Numpy dtype should be float32
+        assert test_data.dtype == np.float32
+        elapsed_time = 0
+        inputs, outputs, bindings, stream = common.allocate_buffers(self.context.engine)
+        result = []
+        for start_idx in range(0, test_data.shape[0], self.batchsize):
+            # If the number of images in the test set is not divisible by the batch size, the last batch will be smaller.
+            # This logic is used for handling that case.
+            end_idx = min(start_idx + self.batchsize, test_data.shape[0])
+            effective_batch_size = end_idx - start_idx
+
+            # Do inference for every batch.
+            inputs[0].host = test_data[start_idx:start_idx + effective_batch_size]
+            t1 = time.time()
+            [output] = common.do_inference_v2(self.context, bindings=bindings, inputs=inputs, outputs=outputs, stream=stream)
+            shape = output.shape[0]
+            output = output[0:int(shape * effective_batch_size / self.batchsize)]
+            elapsed_time += time.time() - t1
+            result.append(output.copy())
+            # Use argmax to get predictions and then check accuracy
+        return result, elapsed_time
```

Comment:
we should add test for this feature. no need to add it into pipeline now, but commit it to repo for manual execution